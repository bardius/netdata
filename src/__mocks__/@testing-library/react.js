import { StylesProvider } from '@material-ui/styles';
import React from 'react';

const react = jest.requireActual('@testing-library/react');

// Mock the MUI component makeStyles className generation method to remove the counter for snapshots
const generateClassName = (rule, styleSheet) => `${styleSheet.options.classNamePrefix}-${rule.key}`;

const render = args => {
  return react.render(
    <StylesProvider generateClassName={generateClassName}>
      {args}
    </StylesProvider>,
  )
}

module.exports = { ...react, render }
