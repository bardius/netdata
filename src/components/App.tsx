import React, { lazy, Suspense } from 'react';
import { Route, Routes } from 'react-router-dom';
import { AppBar, Box, CircularProgress, Container, Toolbar, Typography } from '@material-ui/core';
import { CoinsProvider } from '../contexts/coins/CoinsProvider';
import { Coins } from '../pages/Coins';

const CoinDetailsPage = lazy(() => import(/* webpackChunkName: "CoinDetailsPage" */ '../pages/CoinDetails'));

function App() {
  return (
    <Box data-testid={'app'}>
      <CoinsProvider>
        <Container maxWidth='md'>
          <AppBar position='static' color={'secondary'}>
            <Toolbar variant='dense'>
              <Typography variant='h5' component={'h1'}>
                Netdata Test
              </Typography>
            </Toolbar>
          </AppBar>

          <Suspense
            fallback={
              <Box py={'100px'} width={'100%'}>
                <CircularProgress />
              </Box>
            }>
            <Routes>
              <Route index element={<Coins />} />
              <Route path='coin/:coinMarketId/details' element={<CoinDetailsPage />} />
            </Routes>
          </Suspense>
        </Container>
      </CoinsProvider>
    </Box>
  );
}

export default App;
