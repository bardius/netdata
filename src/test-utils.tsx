import * as React from 'react';
import type { RenderOptions } from '@testing-library/react';
import { render } from '@testing-library/react';
import { CssBaseline, MuiThemeProvider } from '@material-ui/core';
import BASE_THEME from './theme/base';

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <MuiThemeProvider theme={BASE_THEME}>
    <CssBaseline />
    {children}
  </MuiThemeProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options });

export { customRender as render };
