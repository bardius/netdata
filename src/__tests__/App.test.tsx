import React from 'react';
import { screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import App from '../components/App';
import { render } from '../test-utils';

describe('Component - App', () => {
  it('should render with spinner', () => {
    const { asFragment } = render(<BrowserRouter><App /></BrowserRouter>);

    const linkElement = screen.getByRole(/progressbar/i);
    expect(linkElement).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
