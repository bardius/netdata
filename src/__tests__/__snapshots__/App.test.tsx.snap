// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - App should render with spinner 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root"
    data-testid="app"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorSecondary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
        >
          <h1
            class="MuiTypography-root MuiTypography-h5"
          >
            Netdata Test
          </h1>
        </div>
      </header>
      <div
        class="MuiBox-root MuiBox-root"
      >
        <nav
          aria-label="pagination navigation"
          class="MuiPagination-root"
          data-testid="coin-list-pagination"
        >
          <ul
            class="MuiPagination-ul"
          >
            <li>
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textSecondary MuiPaginationItem-disabled MuiPaginationItem-sizeLarge MuiButtonBase-disabled"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiPaginationItem-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  />
                </svg>
              </button>
            </li>
            <li>
              <button
                aria-current="true"
                aria-label="page 1"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textSecondary MuiPaginationItem-selected MuiPaginationItem-sizeLarge"
                tabindex="0"
                type="button"
              >
                1
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </li>
            <li>
              <button
                aria-label="Go to page 2"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textSecondary MuiPaginationItem-sizeLarge"
                tabindex="0"
                type="button"
              >
                2
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </li>
            <li>
              <button
                aria-label="Go to page 3"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textSecondary MuiPaginationItem-sizeLarge"
                tabindex="0"
                type="button"
              >
                3
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </li>
            <li>
              <div
                class="MuiPaginationItem-root MuiPaginationItem-ellipsis MuiPaginationItem-sizeLarge"
              >
                â€¦
              </div>
            </li>
            <li>
              <button
                aria-label="Go to page 10"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textSecondary MuiPaginationItem-sizeLarge"
                tabindex="0"
                type="button"
              >
                10
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </li>
            <li>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textSecondary MuiPaginationItem-sizeLarge"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiPaginationItem-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
        </nav>
        <div
          class="MuiBox-root MuiBox-root"
        >
          <div
            class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
            role="progressbar"
            style="width: 40px; height: 40px;"
          >
            <svg
              class="MuiCircularProgress-svg"
              viewBox="22 22 44 44"
            >
              <circle
                class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                cx="44"
                cy="44"
                fill="none"
                r="20.2"
                stroke-width="3.6"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
